encoding/ssz/query/generalized_index.go

```diff
@@ -0,0 +1,306 @@
+		fieldPos, fieldSsz, err := getContainerFieldByName(currentInfo, element.Name)
+		if err != nil {
+			return 0, fmt.Errorf("container field %s not found: %w", element.Name, err)
+		}
+
+		// Get the chunk count for the current container
+		chunkCount, err := getChunkCount(currentInfo)
+		if err != nil {
+			return 0, fmt.Errorf("chunk count error: %w", err)
+		}
+
+		// Update the generalized index to point to the specified field
+		root = root*nextPowerOfTwo(chunkCount) + fieldPos
+		currentInfo = fieldSsz
+
+		// Check if a path element is a length field
+		if element.Length {
+			currentInfo, root, err = calculateLengthGeneralizedIndex(fieldSsz, element, root)
+			if err != nil {
+				return 0, fmt.Errorf("length calculation error: %w", err)
+			}
+			continue
+		}
+
+		if element.Index == nil {
+			continue
+		}
+
+		switch fieldSsz.sszType {
+		case List:
+			currentInfo, root, err = calculateListGeneralizedIndex(fieldSsz, element, root)
+			if err != nil {
+				return 0, fmt.Errorf("list calculation error: %w", err)
+			}
+
+		case Vector:
+			currentInfo, root, err = calculateVectorGeneralizedIndex(fieldSsz, element, root)
+			if err != nil {
+				return 0, fmt.Errorf("vector calculation error: %w", err)
+			}
+
+		case Bitlist:
+			currentInfo, root, err = calculateBitlistGeneralizedIndex(fieldSsz, element, root)
+			if err != nil {
+				return 0, fmt.Errorf("bitlist calculation error: %w", err)
+			}
+
+		case Bitvector:
+			currentInfo, root, err = calculateBitvectorGeneralizedIndex(fieldSsz, element, root)
+			if err != nil {
+				return 0, fmt.Errorf("bitvector calculation error: %w", err)
+			}
+
+		default:
+			return 0, fmt.Errorf("indexing not supported for type %s", fieldSsz.sszType)
+		}
+
+	}
+
+	return root, nil
+}
+
+// getContainerFieldByName finds a container field by its name
+// and returns its index and SSZInfo.
+func getContainerFieldByName(info *SszInfo, fieldName string) (uint64, *SszInfo, error) {
+	containerInfo, err := info.ContainerInfo()
+	if err != nil {
+		return 0, nil, err
+	}
+
+	for index, name := range containerInfo.order {
+		if name == fieldName {
+			fieldInfo := containerInfo.fields[name]
+			if fieldInfo == nil || fieldInfo.sszInfo == nil {
+				return 0, nil, fmt.Errorf("field %s has no ssz info", name)
+			}
+			return uint64(index), fieldInfo.sszInfo, nil
+		}
+	}
+
+	return 0, nil, fmt.Errorf("field %s not found", fieldName)
+}
+
+// Helpers for Generalized Index calculation per type
+
+// calculateLengthGeneralizedIndex calculates the generalized index for a length field.
+// note: length fields are only valid for List and Bitlist types. Multi-dimensional arrays are not supported.
+func calculateLengthGeneralizedIndex(fieldSsz *SszInfo, element PathElement, root uint64) (*SszInfo, uint64, error) {
+	if element.Index != nil {
+		return nil, 0, fmt.Errorf("len() is not supported for indexed elements (multi-dimensional arrays)")
```

Comment:
```suggestion
		return nil, 0, fmt.Errorf("len() is not supported for multi-dimensional arrays")
```
Would that message be equivalent? Because I find the original one a little hard to understand

Reply:
The original message is very shady. Committed your suggestion!